### PR TITLE
Add a dev doc for resuming SDG runs

### PR DESCRIPTION
**Pull request by bbrowning**
Document the design for how we plan to resume interrupted SDG runs.

### CODE REVIEW

File: docs/sdg/sdg-resume.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+The following types of interruption should be handled:
+
+* user-triggered interruptions (e.g. CTRL-c)
+* SIGINT interruptions (e.g. when running on a spot instance of Pod
+  which receives SIGINT)
+* system hardware failure (GPU failure, power loss)
+
+## Non-Goals
+
+This is not replacing checkpoints. Checkpoints are how we save each
```
Comment (danmcp) [inline docs/sdg/sdg-resume.md:51]: It might be helpful to give the context of how long a batch might take for comparison.

Comment (bbrowning) [reply docs/sdg/sdg-resume.md:51]: Yes, that's a good idea. And I also need to talk about how checkpoints persist across individual data generation runs, while this is focused on the ability to resume a specific interrupted SDG run.

**Comment by bbrowning** [comment]
This is not abandoned, but how we resume SDG runs is going to change as we integrate the context-aware chunking and docling. I was planning to revisit this once we get the former integrated. Just updating status here for visibility and for the stale bot.

**Comment by bbrowning** [comment]
I'm going to close this myself, as SDG has changed so much since this started it will need a complete rethinking.

**Pull request closed without merging**